api/payments.js

```diff
@@ -448,12 +442,15 @@ function getAccountPayments(request, response, next) {
   };
 
   function attachResourceId(transactions, callback) {
-    async.map(transactions, function(payment, async_map_callback) {
+    async.map(transactions, function(paymentResult, async_map_callback) {
+      var hash = paymentResult.hash;
+      var payment = paymentResult.payment;
+
       if (!payment) {
         async_map_callback(new Error('No payment found'));
```

Comment:
Also, I think it might be worthwhile to refactor the `dbinterface` to pass errors to the callback, than do this: https://github.com/ripple/ripple-rest/blob/develop/lib/db-interface.js#L311